modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -835,8 +835,11 @@ class TaskProcessor {
 
                 log.trace "[${safeTaskName(task)}] Cacheable folder=${resumeDir?.toUriString()} -- exists=$exists; try=$tries; shouldTryCache=$shouldTryCache; entry=$entry"
                 final cached = shouldTryCache && exists && entry.trace.isCompleted() && checkCachedOutput(task.clone(), resumeDir, hash, entry)
-                if( cached )
+                if( cached ) {
+                    this.dispatcherClient.createTaskExecution(task.taskId, 0, 'SKIPPED')
```

Comment:
kinda sus to have skipped instead of a more specific status (skips also used for like not taking a branch) but of the existing ones it is the most sensible.